internal/completions/asdf.zsh

```diff
@@ -92,110 +102,243 @@ _asdf__installed_versions_of_plus_system() {
+        if (( CURRENT == 4 )); then
+          _alternative \
+            'flags:flags:((--all\:"Update all installed plugins"))' \
+            'asdf-available-plugins:Installed ASDF Plugins:_asdf__installed_plugins'
+        elif (( CURRENT == 5 )); then
+          if [[ ${words[4]} != "--all" ]]; then
+            local -a refs
+            while IFS=: read -r value descr; do
+              refs+=( "${value}:${descr}" )
+            done < <(_asdf__plugin_git_refs ${words[4]})
+            _describe -V -t git-refs 'Git References' refs
+          fi
+        fi
+        ;;
       (remove)
+        # Complete installed plugin names for remove command
         _asdf__installed_plugins
         return
         ;;
       (list)
-        _asdf__installed_plugins
-        return
-        ;;
-      (*)
-        _describe -t asdf-commands 'ASDF Plugin Commands' asdf_plugin_commands
-        return
+        # Handle list command options with support for --urls and --refs flags
+        case $CURRENT in
+          4)
+            _alternative \
+              'flags:flags:((--urls\:"Show repository URLs" --refs\:"Show Git references"))' \
+              'commands:commands:((all\:"List all available plugins"))'
+            return
+            ;;
+          5)
+            # Handle remaining available flags
+            if [[ ${words[4]} == --* ]]; then
+              local used_flags=("${words[@]}")
+              local -a available_flags
+              available_flags=()
+              if [[ ! "${used_flags[@]}" =~ "--urls" ]]; then
+                available_flags+=("--urls")
+              fi
+              if [[ ! "${used_flags[@]}" =~ "--refs" ]]; then
+                available_flags+=("--refs")
+              fi
+              (( ${#available_flags[@]} )) && compadd -- "${available_flags[@]}"
+            fi
+            return
+            ;;
+        esac
         ;;
-    esac
+      esac
+    fi
     ;;
   (current)
+    # Complete with installed plugins for current command
     _asdf__installed_plugins
     ;;
+  (list)
+    # Handle list command completions with support for 'all' and specific plugins
+    case $CURRENT in
+      3)
+        _alternative \
+          'commands:commands:((all\:"List all available (remote) versions"))' \
+          'plugin:plugin:_asdf__installed_plugins'
+        ;;
+      4)
+        if [[ ${words[3]} == "all" ]]; then
+          _asdf__installed_plugins
+        else
+          # For normal list: show installed versions with optional filter
+          _asdf__installed_versions_of ${words[3]}
+        fi
+        ;;
+      5)
+        # When listing all versions of a specific plugin
+        if [[ ${words[3]} == "all" ]]; then
+          local versions
+          if versions=$(asdf list all "${words[4]}" 2>/dev/null); then
+            _wanted "remote-versions-${words[4]}" \
+              expl "Available versions of ${words[4]}" \
+              compadd -- ${(f)versions}
+          else
```

Comment:
It doesn't really make a ton of sense to suggest specific versions of a tool when the user is typing a `asdf list all` command. It's always going to just print the version the user typed. The second argument is really a filter not a specific version (but it becomes a specific version if the user tab-completes to a full version).

```bash
$ asdf list all ruby 1.8.5-p113
1.8.5-p113
```